mobile/src/card-view.js

```diff
@@ -36,22 +36,34 @@ export class EditCardView extends Component {
       email: props.email || '',
       linkedin: props.linkedin || '',
       twitter: props.twitter || '',
+      isChange: false
     }
   }
 
   // Use the supplied key to assign the value
   onChange = (value, key) => {
-    this.setState({ [key]: value })
+    this.setState({ [key]: value, isChange: true })
   }
 
   // Hand form state over to function passed in through props
   onSave = () => {
-    this.props.updateCard(this.state)
+    const trimmedState = {
+      id: this.props.id,
+      firstName: this.state.firstName.trim(),
```

Comment:
Are all these guaranteed to be non-null and defined?  Otherwise, you'll get null-pointer-exceptions.

Reply:
Yes that have || '' in the state